src/storage/mvcc/reader.rs

```diff
@@ -403,12 +403,15 @@ impl MvccReader {
         }
     }
 
+    // for scan lock request, we must return all locks even if they are generated
```

Comment:
The comment has nothing to do with the implementation?

Reply:
it is a corner case for one dangerous optimazition

Reply:
They are or not generated by the same txn?

Reply:
I will move it to tidb.